### PR TITLE
refactor(cli) Port Register Command

### PR DESCRIPTION
**Pull request by xavdid**
This fixes a long-standing frustration, which is calling `zapier register` with no args. Now title is optional and we'll ask at runtime if we need it. Everything else is the same.

### CODE REVIEW

File: packages/cli/src/utils/command-invocations.js [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
-// returns a promise
-const maybeRegisterApp = async context => {
-  if (!(await hasRegisteredApp())) {
-    context.line(
-      "Looks like this is your first push. Let's register your app on Zapier."
-    );
-    const title = await getInput('Enter app title (Ctrl-C to cancel):\n\n  ');
-    await register(context, title, { printWhenDone: false });
-  }
-};
```
Comment (eliangcs) [inline packages/cli/src/utils/command-invocations.js]: I wonder why is this file removed? Originally we had use cases like calling `maybeRegisterApp` in `zapier push` if the app hasn't been registered. But it looks like we removed those. Maybe there's a reason but I forgot. Can you refresh my memory?

Comment (xavdid) [reply packages/cli/src/utils/command-invocations.js]: good question! During the refactor we pulled that functionality out, so this file wasn't being used. With `registration` being ported, none of the commands this file references exist anymore.

As it is now, commands fail appropriately when auth is wrong or the app isn't registered. So it's not misleading, but not great.

I never loved this previous code approach, so I was going to think through a better way to it in oclif. But for now, we don't need this file anymore.
